server/data/data-library/src/main/java/org/apache/james/rrt/lib/AliasResolverImpl.java

```diff
@@ -0,0 +1,107 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ * http://www.apache.org/licenses/LICENSE-2.0                   *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ***************************************************************/
+
+package org.apache.james.rrt.lib;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import javax.inject.Inject;
+
+import org.apache.james.core.Domain;
+import org.apache.james.core.MailAddress;
+import org.apache.james.core.Username;
+import org.apache.james.rrt.api.AliasResolver;
+import org.apache.james.rrt.api.RecipientRewriteTable;
+import org.apache.james.rrt.api.RecipientRewriteTableException;
+import org.apache.james.util.OptionalUtils;
+import org.apache.james.util.StreamUtils;
+
+import com.github.fge.lambdas.Throwing;
+import com.github.steveash.guavate.Guavate;
+
+public class AliasResolverImpl implements AliasResolver {
+    private final RecipientRewriteTable recipientRewriteTable;
+    private final int mappingLimit;
+
+    @Inject
+    public AliasResolverImpl(RecipientRewriteTable recipientRewriteTable) {
+        this.recipientRewriteTable = recipientRewriteTable;
+        this.mappingLimit = recipientRewriteTable.getConfiguration().getMappingLimit();
```

Comment:
I would rather use a second parameter...

Reply:
I do not get it, what do you have in mind?

Reply:
Well optimally, injecting RecipientRewriteTableConfiguration in both objects.

But I may be cutting hairs in 4...

Reply:
Seems hard to do since RRT is dynamically configured

Reply:
Guice could call the configuration method with the POJO.

We're screwed with Spring.

I agree to keep things as is for now.